src/components/App.js

```diff
@@ -118,13 +118,49 @@ const App = React.createClass({
           maxSize: "80%",
           splitterSize: 1,
           endPanelControl: true,
-          startPanel: this.renderCenterPane(this.props),
+          startPanel: this.renderEditorPane(this.props),
           endPanel: RightSidebar(),
           endPanelCollapsed: this.state.endPanelCollapsed,
-          vert: this.state.vertical
+          vert: this.state.horizontal
         }),
-      })
-    );
+      }));
+  },
+
+  renderVerticalLayout() {
+    return dom.div(
+      { className: "debugger" },
+      SplitBox({
+        style: { width: "100vw" },
+        initialSize: "300px",
+        minSize: 10,
+        maxSize: "50%",
+        splitterSize: 1,
+        vert: this.state.horizontal,
+        startPanel: SplitBox({
+          startPanelCollapsed: this.state.startPanelCollapsed,
+          startPanel: Sources({ sources: this.props.sources }),
+          endPanel: this.renderEditorPane(this.props),
+        }),
+        endPanel: SplitBox({
+          initialSize: "300px",
+          minSize: 10,
+          maxSize: "80%",
+          splitterSize: 1,
+          endPanelControl: true,
+          startPanel: this.renderEditorPane(this.props),
+          endPanel: RightSidebar(),
+          vert: this.state.horizontal
+        }),
+        endPanelCollapsed: this.state.endPanelCollapsed,
+      }));
```

Comment:
hmm, it seems strange that we're rendering the editor panel twice here.

Perhaps we set it up now so that it's

SplitBox (vert):
  - SplitBox (hor):
     - sources 
     - editor
  - RightSidebar

This way we can update the RightSidebar in the next PR